internal/app/api/handlers/order.go

```diff
@@ -0,0 +1,146 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/northmule/gophermart/internal/app/api/rctx"
+	"github.com/northmule/gophermart/internal/app/constants"
+	"github.com/northmule/gophermart/internal/app/repository"
+	"github.com/northmule/gophermart/internal/app/repository/models"
+	"github.com/northmule/gophermart/internal/app/services/logger"
+	orderService "github.com/northmule/gophermart/internal/app/services/order"
+	"io"
+	"net/http"
+	"time"
+)
+
+type OrderHandler struct {
+	manager      repository.Repository
+	orderService *orderService.OrderService
+}
+
+type orderResponse struct {
+	Number     string  `json:"number"`
+	Status     string  `json:"status"`
+	Accrual    float64 `json:"accrual,omitempty"`
+	UploadedAt string  `json:"uploaded_at"`
+}
+
+func NewOrderHandler(manager repository.Repository, orderService *orderService.OrderService) *OrderHandler {
+	instance := &OrderHandler{
+		manager:      manager,
+		orderService: orderService,
+	}
+	return instance
+}
+
+func (o *OrderHandler) UploadingOrder(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		rawBody, err := io.ReadAll(req.Body)
+		if err != nil {
+			logger.LogSugar.Errorf(err.Error())
+			res.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		defer req.Body.Close()
+		if len(rawBody) == 0 {
+			logger.LogSugar.Infof("Пустое тело запроса. Тело запроса должно содержать номер заказа.")
+			res.WriteHeader(http.StatusBadRequest)
+			return
+		}
+		orderNumber := string(rawBody)
+		if !o.orderService.ValidateOrderNumber(orderNumber) {
```

Comment:
Более канонический стиль проверки в Go.
```suggestion
		if valid := o.orderService.ValidateOrderNumber(orderNumber); !valid {
```